public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -68,6 +72,11 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
   const isRecordingRuleType = type === RuleFormType.cloudRecording;
   const isCloudAlertRuleType = type === RuleFormType.cloudAlerting;
 
+  const dispatch_ = useDispatch();
```

Comment:
we need to add this as now we want to check all available data sources with ruler enabled for the smart selector

Reply:
yes, because in the same context we have the dispatch variable from the reducer